docs(README): add react native instructions to readme

fongie opened the pull request:
Using Dinero in our React Native app, we found that no currency symbols were showing on Android. Some digging found that newer versions of react native do not come bundled with Number.toLocaleString by default on Android, but there are instructions in the default react native project build.gradle file on how to include internationalization methods in your build, which fixed the problem for us. I thought it would be nice to include this in the readme, for others such as us who were scratching our heads as to why Dinero was working on iOS but not on Android

## Types of changes

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Compliance

<!-- Is your PR compliant with the contributing guidelines of this project? Make sure you can check all boxes: -->

- [x] My change isn't breaking (it doesn't cause existing functionality to change).
- [x] I have read the [CONTRIBUTING](https://github.com/sarahdayan/dinero.js/blob/develop/CONTRIBUTING.md) document.
- [x] My code follows the coding style of this project.
- [x] I have properly formatted my commit messages with [cz-cli](https://github.com/commitizen/cz-cli), or manually, following the [Angular Commit Messages Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] I have updated the documentation accordingly, or my changes doesn't require documentation changes.
- [x] I have added tests to cover my changes, or my changes doesn't require new tests.
- [x] All new and existing tests pass.

sarahdayan commented on README.md:
```diff
@@ -101,6 +101,10 @@ npm install @types/dinero.js --save
 
+### React Native
+
+Dinero uses `Number.toLocaleString`, which is by default **not** bundled with React Native (0.60+) on Android devices. For formatting and currency symbols to display properly, you need to change the preferred build flavor of JavaScriptCore in your project, by opening `./android/app/build.gradle` and changing the line `def jscFlavor = 'org.webkit:android-jsc:+'` to `def jscFlavor = 'org.webkit:and ...
```
```suggestion
Dinero uses [`Number.prototype.toLocaleString`](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Number/toLocaleString), which by default **isn't bundled with React Native (0.60+) on Android devices**. For formatting and currency symbols to display properly, you need to change the preferred build flavor of JavaScriptCore in your project by opening `./android/app/build.gradle` and changing the line `def jscFlavor = 'org.webkit:android-jsc:+'` to `def jscFlavor = 'org.webkit:android-jsc-intl:+'`.
```

fongie commented:
Ok! I changed my master branch so it contains the develop commits and added my commits to it, and changed the target branch here. Only changing the target branch created lots of conflicts. Hope that works! It is my first contribution :)

sarahdayan commented:
:tada: This PR is included in version 1.8.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/dinero.js/v/1.8.0)
- [GitHub release](https://github.com/dinerojs/dinero.js/releases/tag/v1.8.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
